src/EditorFeatures/Test/Rename/RenameRewriterTests.cs

```diff
@@ -0,0 +1,457 @@
+                }, cancellationToken);
+            await verifier.VerifyDocumentAsync("test1.cs",
+@"
+/// <summary>
+/// <see cref=""Apple2""/> Apple2 is not Lemon2. Lemon2 is not banana2
+/// </summary>
+/// <Lemon2>
+/// </Lemon2>
+public class Apple2
+{
+    // banana2 is not Apple2
+    public void Lemon2(int banana2)
+    {
+        string Apple = ""Apple2, Lemon2 and banana2 are fruit"";
+        string Lemon = $""Apple2, Lemon2 and {banana2} are fruit"";
+    }
+}", cancellationToken);
+        }
+
+        [Fact]
+        public async Task TestCSharpRenameComplexification()
+        {
+            var cancellationToken = CancellationToken.None;
+            using var verifier = new Verifier(@"
+                   <Workspace>
+                       <Project Language=""C#"" CommonReferences=""true"">
+                           <Document FilePath=""test1.cs"">
+<![CDATA[
+public class {|Rename1:Apple|}
+{
+    public void Lemon(int banana)
+    {
+        {|Conflict:{|Rename2:banana2|}|} = 10;
+    }
+    public static int banana2
+    {
+        get; private set;
+    }
+}]]>
+                           </Document>
+                    </Project>
+                </Workspace>
+");
+
+            var renameOption = new SymbolRenameOptions();
+            await verifier.RenameAndAnnotatedDocumentAsync(
+                documentFilePath: "test1.cs",
+                new()
+                {
+                    { "Rename1", ("Apple2", renameOption) },
+                    { "Rename2", ("banana", renameOption) },
+                }, cancellationToken);
+            await verifier.VerifyDocumentAsync("test1.cs",
+@"
+public class Apple2
+{
+    public void Lemon(int banana)
+    {
+        global::Apple2.banana = 10;
+    }
+    public static int banana
+    {
+        get; private set;
+    }
+}", cancellationToken);
+        }
+
+        [Fact]
+        public async Task TestRenameFailed()
+        {
+            var cancellationToken = CancellationToken.None;
+            using var verifier = new Verifier(@"
+                   <Workspace>
+                       <Project Language=""C#"" CommonReferences=""true"">
+                           <Document FilePath=""test1.cs"">
+interface IBar
+{
+    void {|Rename1:Hello|}();
+}
+
+class Bar : IBar
+{
+    public void {|Rename2:Hello|}();
+}
+
+                           </Document>
+                    </Project>
+                </Workspace>
+");
+
+            var renameOption = new SymbolRenameOptions();
+            await Assert.ThrowsAsync<ArgumentException>(() =>
+                verifier.RenameAndAnnotatedDocumentAsync(
+                    documentFilePath: "test1.cs",
+                    new()
+                    {
+                        { "Rename1", ("Hello1", renameOption) },
+                        { "Rename2", ("Hello2", renameOption) },
+                    }, cancellationToken));
+        }
+
+        [Fact]
+        public async Task TestRenameStringFail()
```

Comment:
Can you add a failure test for comments as well?

Reply:
Sure